Sources/SwiftDocC/Model/Rendering/RenderNode/RenderNode+Codable.swift

```diff
@@ -101,6 +96,22 @@ extension RenderNode: Codable {
         try container.encodeIfPresent(diffAvailability, forKey: .diffAvailability)
         try container.encodeIfPresent(variants, forKey: .variants)
         
+        // Encode references if the `.skipsEncodingReferences` value is unset or false.
+        let encodeReferences = (encoder.userInfo[.skipsEncodingReferences] as? Bool) != true
+        
+        if encodeReferences {
+            try container.encode(references.mapValues(CodableRenderReference.init), forKey: .references)
+        }
+        
+        guard encodeReferences || references.isEmpty || variantOverrides != nil else {
```

Comment:
Might be better to wrap the code below the `guard` in an `if` instead, so that we don't accidentally add code below that we want to run regardless of what the conditional evaluates to.